frex/models/explanation_types/contextual_explanation.py

```diff
@@ -0,0 +1,14 @@
+from frex.models import Explanation
+from typing import Any
+from dataclasses import dataclass
+from dataclasses_json import dataclass_json
+
+
+@dataclass_json
+@dataclass(frozen=True)
+class ContextualExplanation(Explanation):
+    """
+    An Explanation based on the context in which a recommendation was produced.
+    """
+
+    based_on_context: Any
```

Comment:
maybe unnecessary to have this explanation model, since the context is already included in candidates... but maybe having this slightly-more-granular type of explanation could be useful.